### PR TITLE
Add FIM Windows Registry wildcard support

### PR DESCRIPTION
**Pull request by Deblintrake09**
|Related issue|
|-------------|
|  #3693            |

## Description

This PR adds  tests support for FIM usage of wildcards in configuration for monitoring Windows registries.

### Added

- Added test suite `test_fim/test_registry/test_registry_wildcards` with a total of 6 test cases.

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @Deblintrake09  (Developer)  | test_fim          | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/39492/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/39588/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/39589/)  | [:green_circle::green_circle::green_circle:](https://github.com/wazuh/wazuh-qa/files/11618926/3693-Manager.zip) | Manager         | https://github.com/wazuh/wazuh-qa/pull/3856/commits/29c29f918935a45b6dbff20acfade73257f8edf2        | Nothing to highlight |
| @Deblintrake09  (Developer)  | test_fim          | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/39490/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/39590/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/39591/) | ⚫⚫⚫ | Linux Agent        |  https://github.com/wazuh/wazuh-qa/pull/3856/commits/29c29f918935a45b6dbff20acfade73257f8edf2       | Nothing to highlight |
| @Deblintrake09  (Developer)  | test_fim           | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/39594/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/39639/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/39640/)  | [:green_circle::green_circle::green_circle:](https://github.com/wazuh/wazuh-qa/files/11605874/3948-Manager.zip) | Windows Agent        |  https://github.com/wazuh/wazuh-qa/pull/3856/commits/29c29f918935a45b6dbff20acfade73257f8edf2       | Nothing to highlight |
| @Deblintrake09  (Developer)  | test_fim    | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/39493/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/39592/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/39593/)| :no_entry_sign: :no_entry_sign: :no_entry_sign: |  Solaris & macOS Agent  | https://github.com/wazuh/wazuh-qa/pull/3856/commits/29c29f918935a45b6dbff20acfade73257f8edf2         | Nothing to highlight |
| @damarisg  (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### CODE REVIEW

File: tests/integration/test_fim/conftest.py [hunk 1]
```diff
@@ -43,6 +43,39 @@ def create_key(request):
                        error_message='Did not receive expected "integrity clear" event')
 
 
+@pytest.fixture(scope='function')
```
Comment (damarisg) [inline tests/integration/test_fim/conftest.py:46]:
```suggestion
@pytest.fixture()
```

File: tests/integration/test_fim/test_registry/test_registry_wildcards/test_registry_wildcards.py [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+                                                t1_configuration_metadata)
+
+t2_configuration_parameters, t2_configuration_metadata, t2_case_ids = get_test_cases_data(t2_cases_path)
+t2_configurations = load_configuration_template(configurations_path, t2_configuration_parameters,
+                                                t2_configuration_metadata)
+
+
+wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+key_name = 'test_key'
+value_name = 'test_value'
```
Comment (damarisg) [inline tests/integration/test_fim/test_registry/test_registry_wildcards/test_registry_wildcards.py:95]: Add a comment that identifies this section.

File: tests/integration/test_fim/test_registry/test_registry_wildcards/test_registry_wildcards.py [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+          when the maximum number of values to monitor has been reached.
+        - Verify that proper FIM events are generated while the database
+          is in 'full database alert' mode.
+
+    input_description: A test case (fim_registry_limit) is contained in external YAML file (wazuh_conf.yaml)
+                       which includes configuration settings for the 'wazuh-syscheckd' daemon. That is combined
+                       with the testing registry key to be monitored defined in this module.
+
+    expected_output:
+        -
```
Comment (damarisg) [inline tests/integration/test_fim/test_registry/test_registry_wildcards/test_registry_wildcards.py:141]: What does it mean?

File: tests/integration/test_fim/test_registry/test_registry_wildcards/test_registry_wildcards.py [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
```
Comment (damarisg) [inline tests/integration/test_fim/test_registry/test_registry_wildcards/test_registry_wildcards.py:216]: What does it mean?

File: tests/integration/test_fim/test_registry/test_registry_wildcards/test_registry_wildcards.py [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+                                               arch='x64')
+    assert event_modified is not None, 'Did not find the expected "registry_value modified" event'
+
+    # Delete the created key and check it's deletion is detected
+    delete_registry_value(reg_handle, value_name)
+    event_deleted = check_registry_crud_event(callback=callback_value_event, path=path, type='deleted', timeout=T_10,
+                                              arch='x64')
+    assert event_deleted is not None, 'Did not find the expected "registry_value deleted" event'
+
+    # Delete key to clean envirmoent
```
Comment (damarisg) [inline tests/integration/test_fim/test_registry/test_registry_wildcards/test_registry_wildcards.py:248]: Fix typo

**Review by damarisg — CHANGES_REQUESTED** [review]

File: deps/wazuh_testing/wazuh_testing/fim.py [hunk 6]
```diff
@@ -1093,6 +1093,38 @@ def callback_detect_integrity_event(line):
...
+    return None
+
+
+def callback_detect_registry_integrity_clear_event(line):
+    event = callback_detect_integrity_event(line)
+    if event and event['component'] == 'fim_registry_key' and event['type'] == 'integrity_clear':
+        return True
+    if event and event['component'] == 'fim_registry_value' and event['type'] == 'integrity_clear':
+        return True
+    return None
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply deps/wazuh_testing/wazuh_testing/fim.py:1125]: This functions were extracted to `modules.fim.event_monitor` and this is adding them back by accident. 
Removed.

File: tests/integration/test_fim/test_registry/test_registry_wildcards/test_registry_wildcards.py (hunk 5 above)
Comment (Deblintrake09) [reply tests/integration/test_fim/test_registry/test_registry_wildcards/test_registry_wildcards.py]: Done in https://github.com/wazuh/wazuh-qa/pull/3856/commits/b50ad9625566811faf8e88847b0fb2542f4bcb69

File: tests/integration/test_fim/test_registry/test_registry_wildcards/test_registry_wildcards.py (hunk 4 above)
Comment (Deblintrake09) [reply tests/integration/test_fim/test_registry/test_registry_wildcards/test_registry_wildcards.py]: Done in https://github.com/wazuh/wazuh-qa/pull/3856/commits/b50ad9625566811faf8e88847b0fb2542f4bcb69

File: tests/integration/test_fim/test_registry/test_registry_wildcards/test_registry_wildcards.py (hunk 2 above)
Comment (Deblintrake09) [reply tests/integration/test_fim/test_registry/test_registry_wildcards/test_registry_wildcards.py:95]: Done in https://github.com/wazuh/wazuh-qa/pull/3856/commits/b50ad9625566811faf8e88847b0fb2542f4bcb69

File: tests/integration/test_fim/conftest.py (hunk 1 above)
Comment (Deblintrake09) [reply tests/integration/test_fim/conftest.py]: Done in https://github.com/wazuh/wazuh-qa/pull/3856/commits/b50ad9625566811faf8e88847b0fb2542f4bcb69

File: deps/wazuh_testing/wazuh_testing/__init__.py [hunk 7]
```diff
@@ -24,6 +24,7 @@
 WAZUH_CONF_PATH = os.path.join(WAZUH_PATH, 'etc', 'ossec.conf')
 WAZUH_LOGS_PATH = os.path.join(WAZUH_PATH, 'logs')
 CLIENT_KEYS_PATH = os.path.join(WAZUH_PATH, 'etc' if platform.system() == 'Linux' else '', 'client.keys')
+CONF_FILE_PATH = os.path.join(WAZUH_PATH, 'etc', 'ossec.conf')
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply deps/wazuh_testing/wazuh_testing/__init__.py]: Removed in https://github.com/wazuh/wazuh-qa/pull/3856/commits/76d3e14deb4b3c2f0c7193cfe77fa204e614ea4e

**Pull request closed without merging**